packages/uniforms/src/joinName.ts

```diff
@@ -1,13 +1,23 @@
 export function joinName(flag: null, ...parts: unknown[]): string[];
 export function joinName(...parts: unknown[]): string;
 export function joinName(...parts: unknown[]) {
-  const name = parts.reduce(
-    (parts: unknown[], part: unknown) =>
-      part || part === 0
-        ? parts.concat(typeof part === 'string' ? part.split('.') : part)
-        : parts,
-    [],
-  );
+  const name: string[] = [];
+  for (let index = 0; index !== parts.length; ++index) {
+    const part = parts[index];
+    if (part || part === 0) {
+      if (typeof part === 'string') {
+        if (part.indexOf('.') !== -1) {
+          name.push(...part.split('.'));
+        } else {
+          name.push(part);
+        }
+      } else if (Array.isArray(part)) {
```

Comment:
[This](https://stackoverflow.com/a/25344701) suggests (should probably double-check though) that this might be even faster with a strict comparison: 

```suggestion
      } else if (Array.isArray(part) === true) {
```

Reply:
Well, this post is from 2014 :smile: Anyway, I've tried it and it didn't change the performance in a measurable way.